### DISCUSSION TITLE
Proper/intended way to create hierarchical menus inside NavigationDrawer

### DISCUSSION THREAD

**Discussion by graealex** [AvaloniaCommunity/Material.Avalonia#214] [Q&A]
What is the proper/intended way to create hierarchical menus inside NavigationDrawer? With MenuItem, I can't get sub-menus to work.

**Comment by SKProCH** [comment]
@appleneko2001 do you have any thoughts?

**Comment by appleneko2001** [comment]
One thing I have mention is `MenuItem` are not usable for Navigation Drawer, they are used for top menu bar, and Context menu related thing.

The closest thing in my thought with Material Design probably something like this: 

![](https://lh3.googleusercontent.com/X764WfgPllVG50Pl52COwAHHDw57K7j4JDG5fE5hByBfBEWj-WCfjVPfsXPlIDEokVgEwmAscsVKM4gCY1zea717FPgGy0nAXn-A=w1064-v0)

source the picture:
https://m2.material.io/components/navigation-drawer#usage

To implement such in with Material.Avalonia, you probably need use something similar to `ListBox` but with grouped items feature and selection feature enabled. Sorry I am too long time not used Avalonia, but this would be my approach to implement this thing.
